operators/pkg/controllers/workspace_controller.go

```diff
@@ -0,0 +1,96 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	crownlabsv1alpha1 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// WorkspaceReconciler reconciles a Workspace object
+type WorkspaceReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=crownlabs.polito.it,resources=workspaces,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=crownlabs.polito.it,resources=workspaces/status,verbs=get;update;patch
+
+// Reconcile reconciles the state of a workspace resource
+func (r *WorkspaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	var ws crownlabsv1alpha1.Workspace
+
+	if err := r.Get(ctx, req.NamespacedName, &ws); err != nil {
+		// reconcile was triggered by a delete request
+		klog.Info(fmt.Sprintf("Workspace %s deleted", req.Name))
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	klog.Info(fmt.Sprintf("Reconciling workspace %s", req.Name))
+
+	nsName := fmt.Sprintf("workspace-%s", ws.Name)
+	ns := v1.Namespace{}
+	ns.Name = nsName
+	ns.Namespace = ""
```

Comment:
Probably the fields could be moved inside the definition (i.e. within the braces)